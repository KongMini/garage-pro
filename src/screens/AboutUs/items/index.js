import {Title} from './Title';
import {Description} from './Description';
import {ContactAt} from './ContactAt';

export {Title, Description, ContactAt};
