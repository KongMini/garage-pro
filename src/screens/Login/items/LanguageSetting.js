import React, {useState} from 'react';
import {View} from 'react-native';

import {TouchableCo, AppText, AppIcon} from '../../../elements';
import {
  CheckLogic,
  ComonStyle,
  Icons,
  IconTypes,
  LanguageService,
  Sizes,
  useAppLanguage,
  useAppTheme,
} from '../../../utils';

/**
 *
 * @param {lable,value} data
 */
const LanguageItem = ({data, onPress, selected}) => {
  const {Colors} = useAppTheme();
  return (
    <TouchableCo
      onPress={onPress}
      style={{paddingTop: Sizes.padding, paddingHorizontal: Sizes.padding}}>
      <AppText
        style={{
          color: selected ? Colors.primary : Colors.text,
          fontWeight: '500',
        }}>
        {data.lable}
      </AppText>
    </TouchableCo>
  );
};
const LanguageSetting = () => {
  const {Strings, setLanguageCode} = useAppLanguage();
  const [selectedLanguage, setSelectedLanguage] = useState(
    LanguageService.getCode(),
  );
  const languages = [
    {
      lable: Strings.Vietnamese,
      value: CheckLogic.Language_code.vietnamese,
    },
    {
      lable: Strings.English,
      value: CheckLogic.Language_code.english,
    },
  ];

  return (
    <View
      style={{
        flexDirection: 'row',
        justifyContent: 'center',
        marginBottom: Sizes.height(8),
      }}>
      {languages.map(item => {
        return (
          <LanguageItem
            key={item.value}
            data={item}
            onPress={() => {
              if (selectedLanguage !== item.value) {
                setSelectedLanguage(item.value);
                setLanguageCode(item.value);
              }
            }}
            selected={selectedLanguage === item.value}
          />
        );
      })}
    </View>
  );
};
export {LanguageSetting};
