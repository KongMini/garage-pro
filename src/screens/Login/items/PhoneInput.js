//import liraries
import React, {useRef} from 'react';
import {View} from 'react-native';
import {Controller, useFormState} from 'react-hook-form';
import PhoneNumberInput from 'react-native-phone-number-input';

import {ComonStyle, Sizes, useAppTheme} from '../../../utils';
import {AppText} from '../../../elements';

export function PhoneInput({
  control,
  name,
  label,
  defaultValue = '',
  rules,
  placeholder,
  containerStyle,
  defaultCode = 'SG',
}) {
  const {errors} = useFormState({control, name});
  const {Colors} = useAppTheme();

  const phoneRef = useRef(null);

  return (
    <View
      style={[
        {paddingHorizontal: Sizes.padding * 2},
        ComonStyle.center,
        containerStyle,
      ]}>
      {!!label && (
        <AppText style={{paddingBottom: Sizes.padding / 4}}>{label}</AppText>
      )}
      <Controller
        defaultValue={defaultValue}
        name={name}
        control={control}
        rules={rules}
        render={({field: {onChange, value}}) => {
          return (
            <PhoneNumberInput
              ref={phoneRef}
              defaultCode={defaultCode}
              onChangeText={onChange}
              value={value}
              placeholderTextColor={Colors.placeholder}
              placeholder={placeholder}
              // disableArrowIcon
              // disabled
              withShadow
              flagButtonStyle={{alignSelf: 'center'}}
              containerStyle={{width: '100%'}}
            />
          );
        }}
      />
      {errors[name] && errors[name].message && (
        <AppText
          style={{
            color: Colors.error,
            paddingTop: Sizes.padding / 4,
            fontSize: Sizes.h6,
            alignSelf: 'flex-start',
          }}>
          {errors[name].message}
        </AppText>
      )}
    </View>
  );
}
