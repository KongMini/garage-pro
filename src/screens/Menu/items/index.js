import {MenuFunction} from './MenuFunction';
import {MenuExtra} from './MenuExtra';

export {MenuFunction, MenuExtra};
