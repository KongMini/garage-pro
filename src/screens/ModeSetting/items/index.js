import {ModeList} from './ModeList';

export {ModeList};
