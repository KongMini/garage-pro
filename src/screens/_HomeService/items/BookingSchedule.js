import React, {useState} from 'react';
import {View, Image, Alert} from 'react-native';

import {KeyboardAwareScrollView} from 'react-native-ui-lib';

import {
  AppButton,
  AppDateInput,
  AppDropdown,
  AppIcon,
  AppInput,
  AppText,
  PickImageWithResize,
} from '../../../elements';
import {
  FetchApi,
  ModalBase,
  Sizes,
  useAppLanguage,
  useAppTheme,
} from '../../../utils';
import {
  useGetDistrict,
  useGetDistrictOld,
  useGetListCity,
  useGetListCityOld,
} from '../../../hooks';
import CheckboxServices from '../modules/CheckboxServices';
import {Controller, useWatch} from 'react-hook-form';

const BookingSchedule = ({
  control,
  setScreen,
  reset,
  getValues,
  setValue,
  handleSubmit,
  watch,
}) => {
  const {Colors} = useAppTheme();
  const {Strings} = useAppLanguage();
  const listCity = useGetListCityOld();
  const {listDistrict} = useGetDistrictOld(control);

  const [submiting, setSubmiting] = useState(false);

  const FIELD = [
    {
      type: 'TEXT_INPUT',
      name: 'address',
      label: Strings.Home_address,
      required: {
        required: {value: true, message: Strings.This_field_is_required},
      },
    },
    {
      type: 'DROPDOWN',
      name: 'city',
      dataDropDown: listCity?.data?.Data || [],
      col: 1,
      required: {
        validate: value =>
          getValues('city') || value ? true : Strings.This_field_is_required,
      },
      keyValueDropdown: 'ten_thanhpho',
      keyResult: 'ma_thanhpho',
      onPressChange: () => setValue('district', ''),
    },
    {
      defaultValue: watch('district'),
      type: 'DROPDOWN',
      name: 'district',
      dataDropDown: listDistrict?.Data || [],
      col: 1,
      styleItemContainer: {left: Sizes.width(52)},
      required: {
        validate: value =>
          getValues('city') || value ? true : Strings.This_field_is_required,
      },
      keyValueDropdown: 'ten_quanhuyen',
      keyResult: 'ma_quanhuyen',
    },
    {
      type: 'DATE_PICKER',
      name: 'date_at',
      label: Strings.Expected_time,
      required: {
        required: {value: true, message: Strings.This_field_is_required},
      },
    },
    {type: 'CHECK_BOX_SERVICE', name: 'service_text'},
  ];

  const onSubmit = async e => {
    setSubmiting(true);
    const service_text = (e.service_text || []).join(', ');
    const item = {...e, service_text: service_text};
    const res = await FetchApi.addHomeService(item);
    setSubmiting(false);
    if (res._msg_code == 1) {
      ModalBase.success('Đặt hẹn thành công');
      reset();
      setValue('address', '');
      setValue('city', '');
      setValue('district', '');
      setValue('note', '');
      setScreen('Information');
    } else {
      ModalBase.error({result: res});
    }
  };

  return (
    <KeyboardAwareScrollView
      contentContainerStyle={{paddingBottom: 200}}
      automaticallyAdjustContentInsets={false}
      showVerticalIndicator={false}>
      <View
        style={{
          marginHorizontal: 30,
          marginTop: Sizes.padding,
          flexDirection: 'row',
          flexWrap: 'wrap',
          justifyContent: 'space-between',
          alignItems: 'flex-start',
        }}>
        {FIELD.map((item, index) => {
          return (
            <View
              key={`${index}`}
              style={{
                justifyContent: 'center',
                marginBottom: Sizes.padding,
                width: item.col === 1 ? '48%' : '100%',
              }}>
              <View>
                {!!item.label && (
                  <AppText
                    style={{
                      marginBottom: 6,
                      color: Colors.greyBold,
                    }}>
                    {item.label}
                    {!!item.required && ' *'}
                  </AppText>
                )}
                {item.type === 'TEXT_INPUT' && (
                  <AppInput
                    defaultValue={item.data}
                    control={control}
                    name={item.name}
                    rules={item.required}
                    containerStyle={{paddingHorizontal: 0, width: '100%'}}
                    inputStyle={{paddingVertical: 10}}
                    style={{
                      borderRadius: Sizes.border_radius,
                      borderColor: Colors.greyLight,
                    }}
                  />
                )}
                {item.type === 'TEXT_AREA' && (
                  <AppInput
                    containerStyle={{paddingHorizontal: 0, width: '100%'}}
                    control={control}
                    name={item.name}
                    rules={item.required}
                    multiline={true}
                    style={{
                      paddingTop: 4,
                      alignItems: 'flex-start',
                      borderRadius: Sizes.border_radius,
                      borderColor: Colors.greyLight,
                      height: 100,
                    }}
                  />
                )}
                {item.type === 'DROPDOWN' && (
                  <AppDropdown
                    key={item.key}
                    control={control}
                    name={item.name}
                    rules={item.required}
                    dataDropDown={item.dataDropDown}
                    defaultValue={item.defaultValue || ''}
                    styleInput={{
                      borderWidth: Sizes.border,
                      borderColor: Colors.greyLight,
                      height: 40,
                      marginTop: 0,
                    }}
                    itemTextStyle={{fontSize: Sizes.h5}}
                    styleValue={{fontSize: Sizes.h5}}
                    styleItemContainer={{
                      width: item.col === 1 ? '40%' : Sizes.width(100) - 61,
                      ...item.styleItemContainer,
                    }}
                    keyValueDropdown={item.keyValueDropdown}
                    keyResult={item.keyResult}
                    onPressChange={item.onPressChange}
                  />
                )}
                {item.type === 'DATE_PICKER' && (
                  <AppDateInput
                    control={control}
                    name={item.name}
                    rules={item.rules}
                    style={{
                      height: 40,
                      borderColor: Colors.greyThin,
                      alignSelf: 'center',
                      marginBottom: Sizes.padding,
                      backgroundColor: Colors.background,
                    }}
                    mode="datetime"
                    maxDate={new Date('2100')}
                  />
                )}
                {item.type === 'CHECK_BOX_SERVICE' && (
                  <CheckboxServices
                    name={item.name}
                    control={control}
                    getValues={getValues}
                  />
                )}
              </View>
              {item.label === Strings.Phone && (
                <View
                  style={{
                    borderBottomWidth: 1,
                    paddingBottom: 16,
                    borderColor: Colors.greyThin,
                    marginHorizontal: 10,
                  }}
                />
              )}
            </View>
          );
        })}
      </View>
      <View
        style={{marginBottom: Sizes.padding, marginHorizontal: Sizes.padding}}>
        <Images control={control} name="file_name" setValue={setValue} />
        <AppInput
          control={control}
          name={'note'}
          multiline={true}
          style={{
            paddingTop: 4,
            alignItems: 'flex-start',
            borderRadius: Sizes.border_radius,
            borderColor: Colors.greyLight,
            height: 100,
          }}
        />
      </View>
      <AppButton
        control={control}
        title={Strings.Booking_2}
        type="primary"
        style={{
          paddingVertical: 4,
          paddingHorizontal: 10,
          alignSelf: 'center',
          marginHorizontal: 0,
          marginRight: Sizes.width(2),
        }}
        onPress={handleSubmit(onSubmit)}
        disabled={submiting}
      />
    </KeyboardAwareScrollView>
  );
};

const Images = ({control, name, setValue}) => {
  const {Strings} = useAppLanguage();
  const {Colors} = useAppTheme();
  const watchAll = useWatch({control});

  const onPressChooseImage = onChange => () => {
    PickImageWithResize.fromGallery(result => {
      try {
        if (result && result.err) {
          if (result.err === 'permission') {
            Alert.alert(
              Strings.Permission_not_granted,
              '',
              [
                {text: Strings.Cancel},
                {
                  text: Strings.Ok,
                  onPress: () => {
                    Linking.openSettings();
                  },
                },
              ],
              {cancelable: true},
            );
            return;
          }
          Alert.alert(result.err);
          return;
        }
        if (result && result.uri) {
          console.log('result.uri', result.uri);
          const list = watchAll[name] || [];
          list.push(result.uri);
          onChange(list);
        }
      } catch (error) {}
    });
  };

  return (
    <>
      <Controller
        control={control}
        name={name}
        render={({field: {onChange, value}}) => (
          <View
            style={{
              paddingHorizontal: Sizes.padding,
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'space-between',
            }}>
            <AppText style={{marginBottom: 6}}>{Strings.Note}</AppText>
            <AppIcon
              disabled={watchAll?.[name]?.length > 2}
              icon="attachment"
              type="Entypo"
              style={{
                textAlign: 'center',
                color: Colors.greybold,
                fontSize: Sizes.h3,
              }}
              hitSlop
              onPress={onPressChooseImage(onChange)}
            />
          </View>
        )}
      />
      <View style={{flexDirection: 'row', paddingHorizontal: 6}}>
        {watchAll?.[name]?.length > 0 &&
          watchAll?.[name].map((item, index) => {
            return (
              <View key={`${index}`} style={{paddingRight: 18}}>
                <Image
                  style={{
                    width: Sizes.width(27.2),
                    height: Sizes.width(27.2),
                    borderRadius: 5,
                    marginBottom: 10,
                    marginTop: 20,
                  }}
                  source={{uri: item}}
                  resizeMode="stretch"
                />
                <AppIcon
                  styleTouch={{
                    backgroundColor: 'white',
                    borderRadius: 100,
                    position: 'absolute',
                    right: 10,
                    top: 10,
                  }}
                  icon={'closecircleo'}
                  type={'AntDesign'}
                  color={Colors.love}
                  size={Sizes.h1}
                  onPress={() => {
                    const list = watchAll[name];
                    list.splice(index, 1);
                    setValue(name, list, {shouldDirty: true});
                  }}
                />
              </View>
            );
          })}
      </View>
    </>
  );
};

export default BookingSchedule;
