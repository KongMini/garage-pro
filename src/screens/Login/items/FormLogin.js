//import liraries
import React, {useState, useRef} from 'react';
import {View} from 'react-native';
import {useForm} from 'react-hook-form';
import {KeyboardAwareScrollView} from 'react-native-ui-lib';

import {
  FetchApi,
  ResetFunction,
  Sizes,
  useAppLanguage,
  useAppTheme,
  AccountService,
  ModalBase,
} from '../../../utils';
import {
  AppButton,
  AppContainer,
  AppImage,
  AppInput,
  AppText,
} from '../../../elements';

import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {LanguageSetting} from './LanguageSetting';
import {useNavigation} from '@react-navigation/native';

export function FormLogin() {
  const {Strings} = useAppLanguage();
  const {Colors} = useAppTheme();
  const insert = useSafeAreaInsets();
  const {navigate} = useNavigation();

  const {control, handleSubmit} = useForm({
    mode: 'all',
  });

  const [submiting, setSubmiting] = useState(false);

  const LOGO = require('../../../utils/images/background_image_login.png');

  const INPUT = [
    {
      label: Strings.Phone_number,
      rules: {required: {value: true, message: Strings.This_field_is_required}},
      name: 'phone',
      keyboardType: 'number-pad',
    },
    {
      label: Strings.Password,
      rules: {required: {value: true, message: Strings.This_field_is_required}},
      name: 'password',
      secureTextEntry: true,
    },
  ];

  const onSubmit = async data => {
    try {
      setSubmiting(true);
      const {password, phone} = data;
      const result = await FetchApi.login({
        username: phone,
        password: password,
      });
      if (result._msg_code === 1) {
        AccountService.set(result._data);
        ResetFunction.resetToHome();
      } else if (
        result.message ===
        'Đăng nhập thất bại. Hãy kiểm tra lại tên đăng nhập và mật khẩu'
      ) {
        ModalBase.error({
          message: Strings.Wrong_login,
        });
      } else if (result.code >= 500) {
        ModalBase.error({
          message: Strings.Server_error,
        });
      } else {
        ModalBase.error({
          message: Strings.Have_an_error + ` (${result.code || 0})`,
        });
      }
    } catch (err) {
      console.log('err', err);
    }
    setSubmiting(false);
  };

  const onForgotPassword = () => {
    navigate('ForgotPassword');
  };
  const onCreateNewAccount = () => {
    navigate('Register');
  };

  return (
    <AppContainer style={{marginTop: insert.top}}>
      <KeyboardAwareScrollView automaticallyAdjustContentInsets={false}>
        <AppImage
          source={LOGO}
          style={{
            width: Sizes.device_width,
            height: (Sizes.device_width * 706) / 1208,
          }}
        />
        <LanguageSetting />

        {INPUT.map(item => (
          <AppInput
            containerStyle={{
              paddingHorizontal: Sizes.padding * 3,
              marginBottom: 10,
            }}
            style={{borderWidth: 0}}
            inputStyle={{
              borderBottomWidth: 1,
              paddingLeft: 0,
              borderColor: Colors.borderColorGrey,
              paddingBottom: 2,
            }}
            eyeStyle={{right: 4, bottom: -14}}
            placeholderTextColor={Colors.borderColorGrey}
            key={item.name}
            placeholder={item.label}
            control={control}
            name={item.name}
            rules={item.rules}
            secureTextEntry={item.secureTextEntry}
            keyboardType={item.keyboardType}
          />
        ))}

        <AppButton
          style={{marginTop: Sizes.padding}}
          control={control}
          title={Strings.Login}
          type="primary"
          onPress={handleSubmit(onSubmit)}
          disabled={submiting}
          // loading={submiting}
          // loading={true}
        />

        <View style={{marginVertical: Sizes.padding * 2.5}}>
          <AppButton
            control={control}
            title={Strings.Forgot_password}
            onPress={onForgotPassword}
          />
        </View>
        <AppButton
          type="primary"
          style={{backgroundColor: '#202680', borderColor: '#202680'}}
          control={control}
          title={Strings.Create_new_account_caron}
          onPress={onCreateNewAccount}
        />
      </KeyboardAwareScrollView>

      <AppText
        style={{
          paddingTop: Sizes.padding,
          fontStyle: 'italic',
          textAlign: 'center',
        }}>
        {Strings.Description_bottom_login}
      </AppText>
    </AppContainer>
  );
}
