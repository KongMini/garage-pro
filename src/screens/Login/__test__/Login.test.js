import 'react-native';

import React from 'react';
import {
  fireEvent,
  render,
  waitFor,
  waitForElementToBeRemoved,
} from '@testing-library/react-native';

import {FormLogin} from '../items';

describe('Testing Form Login', () => {
  test('Render as expert', () => {
    const {toJSON} = render(<FormLogin />);

    expect(toJSON()).toMatchSnapshot();
  });
  test('PhoneIput has no value and Submit button disable by default', async () => {
    const {
      queryByPlaceholderText,
      queryByText,
      getAllByText,
      getByTestId,
      findByText,
    } = render(<FormLogin />);
    const submitButton = queryByText('Login');
    fireEvent(submitButton, 'press');
    expect(submitButton.children[0].children[0]).toEqual('Login');
    expect(submitButton).toBeDisabled();

    const phoneInput = queryByPlaceholderText('Phone Number');

    expect(phoneInput.props.value).toEqual('');
  });
});
