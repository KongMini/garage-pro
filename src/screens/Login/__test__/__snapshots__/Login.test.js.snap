// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Form Login Render as expert 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    getTextInputRefs={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onScroll={[Function]}
    scrollEventThrottle={200}
    scrollToBottomOnKBShow={false}
    scrollToInputAdditionalOffset={75}
    startScrolledToBottom={false}
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 27,
            },
            Object {
              "color": undefined,
              "fontSize": 162,
              "fontWeight": "600",
              "marginVertical": 160,
              "textAlign": "center",
            },
          ]
        }
        testID="title"
      >
        demo
      </Text>
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 64,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flexDirection": "row",
                "width": 600,
              },
              Object {
                "elevation": 10,
                "shadowColor": "rgba(0,0,0,0.4)",
                "shadowOffset": Object {
                  "height": 5,
                  "width": 1,
                },
                "shadowOpacity": 0.34,
                "shadowRadius": 6.27,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "minWidth": 32,
                "opacity": 1,
                "width": 150,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginRight": 10,
                  "width": 30,
                }
              }
            >
              <ActivityIndicator
                size="small"
              />
            </View>
            <Modal
              animated={true}
              animationType="slide"
              hardwareAccelerated={false}
              onDismiss={[Function]}
              onRequestClose={[Function]}
              visible={false}
            >
              <RCTSafeAreaView
                emulateUnlessSupported={true}
                style={
                  Array [
                    Object {
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "#ffffff",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "height": 48,
                          "justifyContent": "center",
                          "width": "15%",
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      nativeID="animatedComponent"
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../react-native-country-picker-modal/lib/assets/images/close.ios.png",
                          }
                        }
                        style={
                          Array [
                            Object {
                              "height": 25,
                              "resizeMode": "contain",
                              "width": 25,
                            },
                            undefined,
                            Object {
                              "tintColor": "#000000",
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                  <TextInput
                    autoCorrect={false}
                    autoFocus={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onClose={[Function]}
                    onFocus={[Function]}
                    placeholder="Enter country name"
                    placeholderTextColor="#aaa"
                    renderFilter={[Function]}
                    style={
                      Array [
                        Object {
                          "height": 48,
                          "width": "70%",
                        },
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 16,
                        },
                      ]
                    }
                    testID="text-input-country-filter"
                    value=""
                    withCloseButton={true}
                    withFilter={true}
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      },
                      Object {
                        "backgroundColor": "#ffffff",
                      },
                    ]
                  }
                >
                  <RCTScrollView
                    ItemSeparatorComponent={[Function]}
                    automaticallyAdjustContentInsets={false}
                    data={Array []}
                    getItem={[Function]}
                    getItemCount={[Function]}
                    getItemLayout={[Function]}
                    initialNumToRender={14}
                    keyExtractor={[Function]}
                    keyboardShouldPersistTaps="handled"
                    onContentSizeChange={[Function]}
                    onLayout={[Function]}
                    onMomentumScrollBegin={[Function]}
                    onMomentumScrollEnd={[Function]}
                    onScroll={[Function]}
                    onScrollBeginDrag={[Function]}
                    onScrollEndDrag={[Function]}
                    onScrollToIndexFailed={[Function]}
                    removeClippedSubviews={false}
                    renderItem={[Function]}
                    scrollEventThrottle={1}
                    stickyHeaderIndices={Array []}
                    testID="list-countries"
                    viewabilityConfigCallbackPairs={Array []}
                  >
                    <View />
                  </RCTScrollView>
                </View>
              </RCTSafeAreaView>
            </Modal>
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAi0lEQVRYR+3WuQ6AIBRE0eHL1T83FBqU5S1szdiY2NyTKcCAzU/Y3AcBXIALcIF0gRPAsehgugDEXnYQrUC88RIgfpuJ+MRrgFmILN4CjEYU4xJgFKIa1wB6Ec24FuBFiHELwIpQxa0ALUId9wAkhCnuBdQQ5ngP4I9wxXsBDyJ9m+8y/g9wAS7ABW4giBshQZji3AAAAABJRU5ErkJggg==",
                }
              }
              style={
                Object {
                  "height": 14,
                  "width": 12,
                }
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#F8F9F9",
                  "flex": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 30,
                  "paddingVertical": 27,
                  "textAlign": "left",
                },
                Object {},
              ]
            }
          >
            <TextInput
              editable={true}
              keyboardAppearance="default"
              keyboardType="number-pad"
              onChangeText={[Function]}
              placeholder="Phone Number"
              selectionColor="black"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "flex": 1,
                    "fontSize": 16,
                  },
                  Object {},
                ]
              }
              value=""
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    animatedStyle={
      Object {
        "value": Object {
          "bottom": 64,
        },
      }
    }
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "rgba(0,0,0,0.26)",
        "borderRadius": 4,
        "bottom": 64,
        "justifyContent": "center",
        "marginBottom": 32,
        "position": "absolute",
        "width": 600,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "paddingVertical": 32,
            "width": 600,
          },
        ]
      }
      testID="AnimatedSubmitButton"
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 27,
            },
            Array [
              Object {
                "fontWeight": "bold",
              },
              Object {
                "color": "rgba(0,0,0,0.54)",
                "fontSize": 36,
              },
            ],
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
